controller/signUpController.js

```diff
@@ -0,0 +1,170 @@
+const User = require('../models/user')
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcrypt')
+const secretKey = require('../config/secretKey')
+const utils = require('../validationUtil/utils')
+const nodemailer = require('nodemailer')
+const dotenv = require('dotenv')
+const http = require('http-status-codes')
+
+dotenv.config({
+  path:'./config/.env'
+})
+
+exports.create = async(req,res)=>{
+    let role = null
+    const user =  await User.findOne({email:req.body.email})
+    if(req.body.user_role == 1) {role="Mentors"}
+    if(req.body.user_role == 0) {role="Students"}
+    const data = {
+        first_name : req.body.first_name,
+        middle_name : req.body.middle_name,
+        last_name : req.body.last_name,
+        email : req.body.email,
+        dob : req.body.dob,
+        address : req.body.address,
+        phone : req.body.phone,
+        user_role : role,
+    }
+    if(user) {
+        return res.status(http.FORBIDDEN).json({status : 'failure',
+            message:http.getStatusText(http.FORBIDDEN),
+            data:data})
+    }
+    else{
+        //hash with salting add random string 
+        bcrypt.hash(req.body.password, 10, async(err,hash) =>{
```

Comment:
@AwaleRohin give a specific name to constants (10)

Reply:
done